src/components/MovieDetails.js

```diff
@@ -1,49 +1,73 @@
 import '../css/MovieDetails.css'
+import React, { Component, Fragment } from 'react'
+import { render } from '@testing-library/react'
+import endpoints from '../endpoints'
 
-const MovieDetails = ({ posters, id }) => {
-  const movie = posters.find(movie => {
-    return movie.id === id
-  })
+class MovieDetails extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      movie: {}
+    }
+  }
 
-  const {
-    title,
-    poster_path,
-    backdrop_path,
-    release_date,
-    overview,
-    genres,
-    budget,
-    revenue,
-    runtime,
-    tagline,
-    average_rating
-  } = movie
+  componentDidMount = () => {
+    console.log(`${endpoints.movies}/${this.props.id}`)
+    fetch(`${endpoints.movies}/${this.props.id}`)
+      .then(response => response.json())
+      .then(data => this.setState({
+        movie: data.movie
+      }))
+      .catch(err => this.setState({
+        error: err.message
+      }))
+  }
 
-  return (
-    <section className="movie-details">
-      <header className="movie-details-header">
-        <h1 className="title">{title}</h1>
-        <h2 className="release">{release_date.split('-')[0]}</h2>
-        <h3 className="runtime">{runtime}</h3>
-        <p className="star">⭐️</p>
-        <h3 className="rating-label">Rating</h3>
-        <h3 className="rating">{average_rating.toFixed(1)}</h3>
-        <p className="dollar-sign1">$</p>
-        <h3 className="budget-label">Budget</h3>
-        <h3 className="budget">{budget}</h3>
-        <p className="dollar-sign2">💲</p>
-        <h3 className="gross-label">Gross</h3>
-        <h3 className="gross">{revenue}</h3>
-      </header>
-      <h3 className="genres">{genres}</h3>
-      <h3 className="tagline">{tagline}</h3>
-      <p className="overview">{overview}</p>
-      <div className="visuals">
-        <img className="poster" src={poster_path} alt={title}/>
-        <img className="trailer" src="https://i.stack.imgur.com/PtbGQ.png" alt="trailer placeholder"/>
-      </div>
-    </section>
-  )
+  render() {
+    const {
+      title,
+      poster_path,
+      backdrop_path,
+      release_date,
+      overview,
+      genres,
+      budget,
+      revenue,
+      runtime,
+      tagline,
+      average_rating
+    } = this.state.movie
```

Comment:
I mentioned this in the previous comment, and I've seen other groups do it for projects, but it is my understanding that it is not the best practice to have anything inside the render method except the return.  I could be (and probably am😆) wrong though!